### PR TITLE
API height menubar macos

### PR DESCRIPTION
**Pull request by shubham2892**
Added an API to get MACOS X menubar height in the screen module using the `[NSApp mainMenu] menuBarHeight` API provided.
#8029

### CODE REVIEW

File: docs/api/screen.md [hunk 1]
```diff
@@ -95,6 +95,12 @@ Returns [`Point`](structures/point.md)
 
 The current absolute position of the mouse pointer.
 
+### `screen.getMenuBarHeight()` _macOS_
+
+Returns [`Integer`] - Height
```
Comment (MarshallOfSound) [inline docs/api/screen.md]: We don't need the "[" and "]" here 👍

Comment (shubham2892) [inline docs/api/screen.md]: Will fix this.

Comment (shubham2892) [inline docs/api/screen.md]: Fixed :)

**Comment by shubham2892** [comment]
Hi @MarshallOfSound, Please let me know if this requires anymore changes. Thank you. 😄

File: atom/browser/api/atom_api_screen.cc [hunk 2]
```diff
@@ -120,6 +120,9 @@ void Screen::BuildPrototype(
       .SetMethod("getPrimaryDisplay", &Screen::GetPrimaryDisplay)
       .SetMethod("getAllDisplays", &Screen::GetAllDisplays)
       .SetMethod("getDisplayNearestPoint", &Screen::GetDisplayNearestPoint)
+      #if defined(OS_MACOSX)
```
Comment (kevinsawicki) [inline atom/browser/api/atom_api_screen.cc]: This should be outdented to the beginning of the line.

Comment (kevinsawicki) [inline atom/browser/api/atom_api_screen.cc]: This should be outdented to the beginning of the line.

File: atom/browser/api/atom_api_screen.cc [hunk 3]
```diff
@@ -120,6 +120,9 @@ void Screen::BuildPrototype(
...
+      .SetMethod("getMenuBarHeight", &Screen::getMenuBarHeight)
+      #endif
```
Comment (kevinsawicki) [inline atom/browser/api/atom_api_screen.cc]: This should be outdented to the beginning of the line.

File: atom/browser/api/atom_api_screen_mac.mm [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#import "atom/browser/api/atom_api_screen.h"
+
+#import <Cocoa/Cocoa.h>
+
+namespace atom {
+
+namespace api {
+#if defined(OS_MACOSX)
```
Comment (kevinsawicki) [inline atom/browser/api/atom_api_screen_mac.mm]: This file will only be compiled on macOS so you don't need this `#if` here.

File: atom/browser/api/atom_api_screen_mac.mm [hunk 5]
```diff
@@ -0,0 +1,15 @@
+#import "atom/browser/api/atom_api_screen.h"
```
Comment (kevinsawicki) [inline atom/browser/api/atom_api_screen_mac.mm]: This file needs a license header, please copy it from another file.

File: atom/browser/api/atom_api_screen_mac.mm [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+#import <Cocoa/Cocoa.h>
+
+namespace atom {
+
+namespace api {
+#if defined(OS_MACOSX)
+int Screen::getMenuBarHeight(){
```
Comment (kevinsawicki) [inline atom/browser/api/atom_api_screen_mac.mm]: Space needed between `)` and `{`

File: filenames.gypi [hunk 7]
```diff
@@ -135,6 +135,7 @@
       'atom/browser/api/atom_api_protocol.h',
       'atom/browser/api/atom_api_render_process_preferences.cc',
       'atom/browser/api/atom_api_render_process_preferences.h',
+      'atom/browser/api/atom_api_screen_mac.mm',
```
Comment (kevinsawicki) [inline filenames.gypi]: This line should be added after `'atom/browser/api/atom_api_screen.h',` to preserve the correct ordering.

File: atom/browser/api/atom_api_screen.cc [hunk 8]
```diff
@@ -120,6 +120,9 @@ void Screen::BuildPrototype(
       .SetMethod("getPrimaryDisplay", &Screen::GetPrimaryDisplay)
       .SetMethod("getAllDisplays", &Screen::GetAllDisplays)
       .SetMethod("getDisplayNearestPoint", &Screen::GetDisplayNearestPoint)
+      #if defined(OS_MACOSX)
+      .SetMethod("getMenuBarHeight", &Screen::getMenuBarHeight)
```
Comment (kevinsawicki) [inline atom/browser/api/atom_api_screen.cc]: Please add a simple test for this method that verifies it returns a number to `spec/api-screen-spec.js`

**Comment by shubham2892** [comment]
Hi @kevinsawicki, addressed the comments left by you. Thanks.

**Comment by kevinsawicki** [comment]
Thanks @shubham2892, great job on this 👍 🚀

**Pull request merged**